code/modules/mob/living/simple_animal/hostile/mining/elites/legionnaire.dm

```diff
@@ -0,0 +1,337 @@
+	chosen_attack_num = BONFIRE_TELEPORT
+
+/datum/action/innate/elite_attack/spew_smoke
+	name = "Spew Smoke"
+	button_icon_state = "spew_smoke"
+	chosen_message = "<span class='boldwarning'>Your head will spew smoke in an area, wherever it may be.</span>"
+	chosen_attack_num = SPEW_SMOKE
+
+/mob/living/simple_animal/hostile/asteroid/elite/legionnaire/OpenFire()
+	if(client)
+		switch(chosen_attack)
+			if(LEGIONNAIRE_CHARGE)
+				legionnaire_charge(target)
+			if(HEAD_DETACH)
+				head_detach(target)
+			if(BONFIRE_TELEPORT)
+				bonfire_teleport()
+			if(SPEW_SMOKE)
+				spew_smoke()
+		return
+	var/aiattack = rand(1,4)
+	switch(aiattack)
+		if(LEGIONNAIRE_CHARGE)
+			legionnaire_charge(target)
+		if(HEAD_DETACH)
+			head_detach(target)
+		if(BONFIRE_TELEPORT)
+			bonfire_teleport()
+		if(SPEW_SMOKE)
+			spew_smoke()
+
+/mob/living/simple_animal/hostile/asteroid/elite/legionnaire/Move()
+	if(charging)
+		return FALSE
+	return ..()
+
+/mob/living/simple_animal/hostile/asteroid/elite/legionnaire/MiddleClickOn(atom/A)
+	. = ..()
+	if(!myhead)
+		return
+	var/turf/T = get_turf(A)
+	if(T)
+		myhead.LoseTarget()
+		myhead.Goto(T, myhead.move_to_delay)
+
+/mob/living/simple_animal/hostile/asteroid/elite/legionnaire/proc/legionnaire_charge(target)
+	ranged_cooldown = world.time + 4.0 SECONDS * revive_multiplier()
+	charging = TRUE
+	var/dir_to_target = get_dir(get_turf(src), get_turf(target))
+	var/turf/T = get_step(get_turf(src), dir_to_target)
+	for(var/i in 1 to 6)
+		new /obj/effect/temp_visual/dragon_swoop/legionnaire(T)
+		T = get_step(T, dir_to_target)
+	playsound(src, 'sound/misc/demon_attack1.ogg', 200, 1)
+	visible_message("<span class='danger'>[src] prepares to charge!</span>")
+	addtimer(CALLBACK(src, .proc/legionnaire_charge_2, dir_to_target, 0), 4)
+
+/mob/living/simple_animal/hostile/asteroid/elite/legionnaire/proc/legionnaire_charge_2(move_dir, times_ran)
+	if(times_ran >= 6)
+		charging = FALSE
+		return
+	var/turf/T = get_step(get_turf(src), move_dir)
+	if(ismineralturf(T))
+		var/turf/simulated/mineral/M = T
+		M.gets_drilled()
+	if(T.density)
+		charging = FALSE
+		return
+	for(var/obj/structure/window/W in T.contents)
+		charging = FALSE
+		return
+	for(var/obj/machinery/door/D in T.contents)
+		if(D.density)
+			charging = FALSE
+			return
+	forceMove(T)
+	playsound(src,'sound/effects/bang.ogg', 200, 1)
+	var/list/hit_things = list()
+	var/throwtarget = get_edge_target_turf(src, move_dir)
+	for(var/mob/living/L in T.contents - hit_things - src)
+		if(faction_check_mob(L))
+			return
+		hit_things += L
+		visible_message("<span class='danger'>[src] tramples and kicks [L]!</span>")
+		to_chat(L, "<span class='userdanger'>[src] tramples you and kicks you away!</span>")
+		L.throw_at(throwtarget, 10, 1, src)
+		L.KnockDown(1 SECONDS)
+		L.adjustBruteLoss(melee_damage_upper)
+	addtimer(CALLBACK(src, .proc/legionnaire_charge_2, move_dir, (times_ran + 1)), 0.7)
```

Comment:
does a decimal decisecond timer even work?

Reply:
I'm not sure if it acts the same as 1 decisecond or not, to be honest